sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ModelsTest.java

```diff
@@ -0,0 +1,133 @@
+            });
+        });
+
+        for (int modelIndex = 0; modelIndex < createdModels.size(); modelIndex++) {
+            final ModelData expected = createdModels.get(modelIndex);
+
+            // Get the model
+            getModelRunner(expected.getId(), (modelId) -> {
+                ModelData actual = client.getModel(modelId);
+                assertModelDataAreEqual(expected, actual, false);
+                logger.info("Model {} matched expectations", modelId);
+            });
+
+            // Decommission the model
+            decommissionModelRunner(expected.getId(), (modelId) -> {
+                logger.info("Decommissioning model {}", modelId);
+                client.decommissionModel(modelId);
+            });
+
+            // Get the model again to see if it was decommissioned as expected
+            getModelRunner(expected.getId(), (modelId) -> {
+                ModelData actual = client.getModel(modelId);
+                assertTrue(actual.isDecommissioned());
+                logger.info("Model {} was decommissioned successfully", modelId);
+            });
+
+            // Delete the model
+            deleteModelRunner(expected.getId(), (modelId) -> {
+                logger.info("Deleting model {}", modelId);
+                client.deleteModel(modelId);
+            });
+        }
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.digitaltwins.core.TestHelper#getTestParameters")
+    @Override
+    public void getModelThrowsIfModelDoesNotExist(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) {
+        DigitalTwinsClient client = getClient(httpClient, serviceVersion);
+        final String nonExistantModelId = "urn:doesnotexist:fakemodel:1000";
+        getModelRunner(nonExistantModelId, (modelId) -> {
+            assertRestException(() -> client.getModel(modelId), HttpURLConnection.HTTP_NOT_FOUND);
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.digitaltwins.core.TestHelper#getTestParameters")
+    @Override
+    public void createModelThrowsIfModelAlreadyExists(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) {
+        DigitalTwinsClient client = getClient(httpClient, serviceVersion);
+
+        final List<String> modelsToCreate = new ArrayList<>();
+        final String wardModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.WARD_MODEL_ID, client, randomIntegerStringGenerator);
+        final String wardModelPayload = TestAssetsHelper.getWardModelPayload(wardModelId);
+        modelsToCreate.add(wardModelPayload);
+
+        List<ModelData> createdModels = client.createModels(modelsToCreate);
+        createdModels.forEach((modelData) -> {
+            assertNotNull(modelData);
+        });
+
+        assertRestException(
+            () -> client.createModels(modelsToCreate).forEach((modelData) -> {
```

Comment:
we might not to do this for-each iteration, create models does not return a pageable anymore, so the blocking call should directly complete with result.